crates/turborepo-lib/src/cli/mod.rs

```diff
@@ -758,6 +761,192 @@ impl RunArgs {
+            telemetry.track_arg_usage("experimental-space-id", true);
+        }
+
+        // track values
+        if let Some(dry_run) = &self.dry_run {
+            telemetry.track_arg_value(
+                "dry-run",
+                match dry_run {
+                    DryRunMode::Text => "text",
+                    DryRunMode::Json => "json",
+                },
+                EventType::NonSensitive,
+            );
+        }
+
+        if self.cache_workers != DEFAULT_NUM_WORKERS {
+            telemetry.track_arg_value("cache-workers", self.cache_workers, EventType::NonSensitive);
+        }
+
+        if let Some(concurrency) = &self.concurrency {
+            telemetry.track_arg_value("concurrency", concurrency, EventType::NonSensitive);
+        }
+
+        if !self.global_deps.is_empty() {
+            telemetry.track_arg_value("global-deps", self.cache_workers, EventType::NonSensitive);
+        }
+
+        if let Some(graph) = &self.graph {
+            telemetry.track_arg_value("graph", graph, EventType::NonSensitive);
+        }
+
+        if self.env_mode != EnvMode::default() {
+            telemetry.track_arg_value(
+                "env-mode",
+                match self.env_mode {
+                    EnvMode::Infer => "infer",
+                    EnvMode::Loose => "loose",
+                    EnvMode::Strict => "strict",
+                },
+                EventType::NonSensitive,
+            );
+        }
+
+        if let Some(output_logs) = &self.output_logs {
+            telemetry.track_arg_value(
+                "output-logs",
+                match output_logs {
+                    OutputLogsMode::Full => "full",
+                    OutputLogsMode::None => "none",
+                    OutputLogsMode::HashOnly => "hash-only",
+                    OutputLogsMode::NewOnly => "new-only",
+                    OutputLogsMode::ErrorsOnly => "errors-only",
+                },
+                EventType::NonSensitive,
+            );
+        }
+
+        if self.log_order != LogOrder::default() {
+            telemetry.track_arg_value(
+                "log-order",
+                match self.log_order {
+                    LogOrder::Auto => "auto",
+                    LogOrder::Stream => "stream",
+                    LogOrder::Grouped => "grouped",
+                },
+                EventType::NonSensitive,
+            );
+        }
+
+        if self.log_prefix != LogPrefix::default() {
+            telemetry.track_arg_value(
+                "log-prefix",
+                match self.log_prefix {
+                    LogPrefix::Auto => "auto",
+                    LogPrefix::None => "none",
+                    LogPrefix::Task => "task",
+                },
+                EventType::NonSensitive,
+            );
+        }
+
+        // track sizes
+        if !self.filter.is_empty() {
+            telemetry.track_arg_value("filter", self.filter.len(), EventType::NonSensitive);
```

Comment:
I checked out our [docs](https://turbo.build/repo/docs/core-concepts/monorepos/filtering) for all the filtering options. Could we have the telemetry broken down by each style? (which probably means we have to do this tracking _after_ we've parsed it)

| Filter type | logic |
| --- | --- |
| Filter by workspace name | alphanumeric value
| Filter by workspace directory | starts with `./`
| Include dependents of matched workspaces | starts with `...`
| Include dependencies of matched workspaces | ends with `...`
| Execute tasks from the workspace root | matches `//`
| Filter by changes in git history | wrapped in `[]`
| Exclude workspaces from selection | starts with `!`

I would also _love_ to break down by `*` values in the first two (e.g. `-F=util-*` or `-F=./packages/*`), but I've excluded that here.

I'd propose just making the _value_ of this `filter` arg some enum that corresponds to each of these as they _should_ be mutually exclusive.

Reply:
Yea agreed I mentioned that above as well, we can add other events in the filter parser itself to track all of these - I have an task for it!